wp-remember-the-galleries/wp-remember-the-galleries.php

```diff
@@ -308,50 +313,74 @@ static function save_gallery() {
 			wp_send_json_error( 'empty-name' );
 		}
 
+		$term_id = null; 
+		if ( isset( $_POST['term_id'] ) ) {
+			$term_id = (int) $_POST['term_id'];
+		}
+
 		$gallery_name = sanitize_text_field( trim( $_POST['name'] ) );
 
 		if( empty( $gallery_name ) ) {
 			wp_send_json_error( 'empty-name' );
 		}
 
-		// Get or insert the appropriate term for this gallery
+		// Does a gallery with this name already exist?
 		$term_info = term_exists( $gallery_name, self::entity );
+		$allow_rename = true;
+		if ( isset( $term_info['term_id'] ) && $term_info['term_id'] != $term_id ) {
+			// Ask user to confirm that they want to overwrite the existing gallery
+			if ( !isset( $_POST['yes'] ) ) {
+				wp_send_json_error( 'need-confirm' );
+			}
+			// User has confirmed they want to overwrite the existing gallery, so let's use that term id
+			else {
+				$allow_rename = false;
+				$term_id = $term_info['term_id'];
+			}
+		}
 
-		if( !isset( $term_info['term_id'] ) ) {
-			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '--' ) );
+		if( !$term_id ) {
+			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '-' . $post_id ) );
 
 			if( is_wp_error( $term_info ) ) {
-				wp_send_json_error( $term_info );
+				wp_send_json_error( $term_info->get_error_message() );
 			}
-		}
-
-		$term = get_term( $term_info['term_id'], self::entity );
 
-		// Get the ID out of the term slug:
-		if( preg_match( '/^' . self::entity . '-(\d+)$/', $term->slug, $matches ) ) {
-			$post = get_post( (int) $matches[1] );
+			$term_id = $term_info['term_id'];
+		}
 
-			// Ensure this is the right kind of post. If not, just create a new one.
-			if( $post && $post->post_type == self::entity ) {
-				$post_id = $post->ID;
-			}
+		if( $term_id && $allow_rename ) {
+			wp_update_term( $term_id, self::entity, array( 'name' => $gallery_name ) ); 
 		}
 
-		if( !isset( $post_id ) ) {
+		// Get the post ID associated with this term
+		$post_id = self::get_post_id( $term_id );
+		
+		if( !$post_id ) {
 			$post_id = wp_insert_post( array(
 				'post_title' => $gallery_name,
 				'post_type' => self::entity,
 				'post_status' => 'publish'
 			) );
 
-			wp_update_term( $term->term_id, $term->taxonomy, array( 'slug' => self::entity . '-' . $post_id ) );
```

Comment:
Here is where the post is updated with the slug now that the post is known.

The reason that the term is going in first is because there is a key on the title for that term ((taxonomy, slug) in 4.3+) – and it's faster to check against that and validate that the gallery name is unique. This is used to limit the post to one.

So what's happening now is records like this are being created in wp_terms:
```
|   21581 | I am a new gallery!                              | wp_rtg-3         |          0 |          0 |
|   21580 | I need a name!                                   | wp_rtg-2         |          0 |          0 |
|   21579 | I'm going to name this thus-far un-named gallery | wp_rtg           |          0 |          0 |
```